src/main/java/com/cloudbees/jenkins/plugins/sshagent/exec/ExecRemoteAgent.java

```diff
@@ -24,76 +24,58 @@
 
 package com.cloudbees.jenkins.plugins.sshagent.exec;
 
-import com.cloudbees.jenkins.plugins.sshagent.LauncherProvider;
-import com.cloudbees.jenkins.plugins.sshagent.RemoteAgent;
 import hudson.AbortException;
 import hudson.FilePath;
+import hudson.Launcher;
 import hudson.model.TaskListener;
+import hudson.slaves.WorkspaceList;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.Serializable;
 import java.nio.charset.StandardCharsets;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
 
-
 /**
- * An implementation that uses native SSH agent installed on a system.
+ * Runs a native SSH agent installed on a system.
  */
-public class ExecRemoteAgent implements RemoteAgent {
+public final class ExecRemoteAgent implements Serializable {
```

Comment:
OK to serialize now that this merely contains a `Map<String, String>`, and means that the `StepExecution` does not need to recreate it.